caffe2/CMakeLists.txt

```diff
@@ -1529,7 +1529,20 @@ if(USE_XPU)
   target_link_libraries(
       torch_xpu PRIVATE ${Caffe2_XPU_DEPENDENCY_LIBS})
 
-  target_link_libraries(torch_xpu PUBLIC torch_cpu_library)
+  include(CheckLinkerFlag)
+
+  # Check whether the compiler supports '--no-as-needed' and '--as-needed'
+  check_linker_flag(CXX "-Wl,--no-as-needed,-Wl,--as-needed" HAVE_AS_NEEDED)
```

Comment:
```suggestion
  check_linker_flag(CXX "-Wl,--no-as-needed -Wl,--as-needed" HAVE_AS_NEEDED)
```

Reply:
@EikanWang The delimiter has been fixed with `;`.

Reply:
`;` is suspicious. `HAVE_AS_NEEDED` would be always `FALSE` because compiler cannot support `-Wl,--no-as-needed;-Wl,--as-needed`. Please double check.

Reply:
`;` has been checked locally and `check_linker_flag` doesn't work with the delimiter ` `.
To avoid potential delimiter issues caused by different cmake versions or other factors, two linker flags will be checked separately.